notebooks/algorithm-design/concepts.ipynb

```diff
@@ -0,0 +1,544 @@
+      "text/plain": [
+       "<Figure size 417.879x144.48 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "from qiskit import QuantumCircuit\n",
+    "from qiskit.quantum_info import SparsePauliOp\n",
+    "\n",
+    "# The following code will work for any other initial single-qubit state and observable\n",
+    "original_circuit = QuantumCircuit(1)\n",
+    "original_circuit.h(0)\n",
+    "\n",
+    "H = SparsePauliOp([\"X\", \"Z\"], [2, -1])\n",
+    "\n",
+    "aux_circuits = []\n",
+    "for pauli in H.paulis:\n",
+    "    aux_circ = original_circuit.copy()\n",
+    "    aux_circ.barrier()\n",
+    "    if str(pauli) == \"X\":\n",
+    "        aux_circ.h(0)\n",
+    "    elif str(pauli) == \"Y\":\n",
+    "        aux_circ.sdg(0)\n",
+    "        aux_circ.h(0)\n",
+    "    else:\n",
+    "        aux_circ.i(0)\n",
+    "    aux_circ.measure_all()\n",
+    "    aux_circuits.append(aux_circ)\n",
+    "\n",
+    "\n",
+    "print(\"Original circuit:\")\n",
+    "display(original_circuit.draw(\"mpl\"))\n",
+    "for (circuit, pauli) in zip(aux_circuits, H.paulis):\n",
+    "    print(f\"Auxiliary circuit for {str(pauli)}\")\n",
+    "    display(circuit.draw(\"mpl\"))"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
```

Comment:
Can this code be turned interactive? Can we this code up into steps and have the reader either a) add code or b) fix incorrect code to do this themselves? Can we break this cell into multiple small cells for each task? (import statements, initialize state, create measurement circuits...)

Maybe call them "measurement circuits" instead of "auxiliary". Can each chunk of code be commented? 

Basically, do you hope that the reader will know how to do these sorts of tasks after reading this section? If so, more scaffolding will likely be required.